vsphere/distributed_virtual_switch_structure.go

```diff
@@ -0,0 +1,384 @@
+	}
+}
+
+func schemaDistributedVirtualSwitchHostMemberConfigSpec() *schema.Schema {
+	se := map[string]*schema.Schema{
+		"max_proxy_switch_ports": &schema.Schema{
+			Type:         schema.TypeInt,
+			Optional:     true,
+			Description:  "Maximum number of ports allowed in the HostProxySwitch.",
+			ValidateFunc: validation.IntAtLeast(0),
+		},
+		// The host name should be enough to get a reference to it, which is what we need here
+		"host_system_id": &schema.Schema{
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "The managed object ID of the host to search for NICs on.",
+		},
+		"operation": &schema.Schema{
+			Type:         schema.TypeInt,
+			Optional:     true,
+			Description:  "Host member operation type.",
+			ValidateFunc: validation.StringInSlice(configSpecOperationAllowedValues, false),
+		},
+		// DistributedVirtualSwitchHostMemberPnicBacking extends DistributedVirtualSwitchHostMemberBacking
+		// which is a base class
+		"backing":                schemaDistributedVirtualSwitchHostMemberPnicBacking(),
+		"vendor_specific_config": schemaDistributedVirtualSwitchKeyedOpaqueBlob(),
+	}
+
+	s := &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: se,
+		},
+	}
+
+	return s
+}
+
+func expandDistributedVirtualSwitchHostMemberConfigSpec(d *schema.ResourceData, dvs *mo.DistributedVirtualSwitch, refs map[string]types.ManagedObjectReference) []types.DistributedVirtualSwitchHostMemberConfigSpec {
+	// Configure the host and nic cards used as uplink for the DVS
+	var hmc []types.DistributedVirtualSwitchHostMemberConfigSpec
+
+	if hosts, ok := d.GetOk("host"); ok {
+		hosts := hosts.([]interface{})
+		// If the DVS exist we go through all the hosts and see which ones
+		// we have to delete or modify
+		if dvs != nil {
+			config := dvs.Config.GetDVSConfigInfo()
+			for _, h := range config.Host {
+				if host := isHostPartOfDVS(hosts, refs, h.Config.Host); host != nil {
+					// Edit
+					backing := new(types.DistributedVirtualSwitchHostMemberPnicBacking)
+					for _, nic := range host["backing"].([]interface{}) {
+						backing.PnicSpec = append(backing.PnicSpec, types.DistributedVirtualSwitchHostMemberPnicSpec{
+							PnicDevice: strings.TrimSpace(nic.(string)),
+						})
+					}
+					hcs := types.DistributedVirtualSwitchHostMemberConfigSpec{
+						Host:      *h.Config.Host,
+						Backing:   backing,
+						Operation: "edit", // Options: "add", "edit", "remove"
+					}
+					hmc = append(hmc, hcs)
+
+					// We take it out from the refs, on the last pass we consider whatever
+					// is left as to be added
+					delete(refs, host["host_system_id"].(string))
+				} else {
+					// Remove
+					// XXX I'm not sure if it's necessary to mention the specific NICs when removing a host completely
+					backing := new(types.DistributedVirtualSwitchHostMemberPnicBacking)
+					cbp := h.Config.Backing.GetDistributedVirtualSwitchHostMemberBacking()
```

Comment:
I got a box with 4 NICs to test this... doesn't work :( . I'm trying to figure it out what's wrong.

Reply:
I haven't been able so far to retrieve the ```PnicSpec``` this way, will keep on trying.

Reply:
That said taking out the backing here (remove lines 151 to 158 and line 161) seems to work, although I still need a way to get this info for ```flattenDistributedVirtualSwitchHostMemberConfigSpec```.

Reply:
I test it further and realised that creating and modifying uplinks (either adding or removing) works just fine but completely removing a host from a dvs doesn't work (my previous comment was wrong). It feels like I'm a inch away from getting that info through this structure but it's not going to be possible. I will check if I can get it somehow through the ```HostSystem``` reference.